lib/commands/command-plugin-update.bash

```diff
@@ -28,11 +27,13 @@ plugin_update_command() {
 update_plugin() {
   local plugin_name=$1
   local plugin_path=$2
-  local gitref=$3
+  repo_default_branch=$(git --git-dir "$plugin_path/.git" --work-tree "$plugin_path" ls-remote --symref origin HEAD | awk -F'[/\t]' 'NR == 1 {print $3}')
```

Comment:
This is the magic as `ls-remote` calls out to the provided remote git server to get the refs instead of relying on the current checked-out state of the repo.

I am yet to write tests for this that allow `ls-remote` to work (pretty sure we can achieve this with a local 'remote')